doc/BorderRouter.rst

```diff
@@ -1,182 +1,220 @@
-# Border Router Redesign
+**********************************************
+Border Router Performance Optimized Redesign
+**********************************************
 
-* Author: Justin Rohrer
-* Last Updated: 2023-04-12
-* Status: In Review
-* Discussion at: [#4334](https://github.com/scionproto/scion/issues/4334)
-* Implemented in: [](will be added once implemented)
+- Author: Justin Rohrer
+- Last Updated: 2023-04-28
+- Discussion at: `#4334 <https://github.com/scionproto/scion/issues/4334>`_
 
-## Motivation
+Abstract
+===========
 
 Right now, the performance of the border router is very much limited because a single goroutine per
 border router interface is responsible for reading, parsing, processing and forwarding the packets.
+
+Background
+===========
+
 Redesigning this pipeline is expected to lead to significant performance improvements.
 The current design was not created with performance in mind.
 Previously, there also existed a design which did something similar than the proposed design in
 this document.
-[See](https://github.com/scionproto/scion/tree/92531f5cb62197b9d705001c13e5a6bdb7ba1fa4/go/border).
+`See <https://github.com/scionproto/scion/tree/92531f5cb62197b9d705001c13e5a6bdb7ba1fa4/go/border>`_.
 
-## Overview
+Proposal
+========
 
 The pipeline gets changed to have seperate goroutines for the receiving, processing and forwarding steps.
```

Comment:
Also mention that we don't need to allocate memory at runtime anymore.
It is all preallocated